### PR TITLE
[DPP-954] Allow pruning up to the ledger end offset

### PR DESCRIPTION
**Pull request by pbatko-da**
Before the pruning offset was required to be before the ledger end offset. Now the ledger end offset becomes a valid pruning offset

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiParticipantPruningService.scala [hunk 1]
```diff
@@ -185,8 +185,7 @@ final class ApiParticipantPruningService private (
     for {
       ledgerEnd <- readBackend.currentLedgerEnd()
       _ <-
-        // NOTE: This constraint should be relaxed to (pruneUpToString <= ledgerEnd.value)
-        if (pruneUpToString < ledgerEnd.value) Future.successful(())
+        if (pruneUpToString <= ledgerEnd.value) Future.successful(())
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiParticipantPruningService.scala:188]: you probably need to adapt the conformance test in `ParticipantPruningIT`

**Comment by tudor-da** [comment]
Thanks. Additionally, we would need to:
- [ ] adapt the docs where we mention this restriction at least here (https://docs.daml.com/ops/pruning.html#determine-a-suitable-pruning-offset)
- [ ] Remove the redundant insertions in the conformance tests where we do pruning (that were used to ensure pruning before ledger end).

**Comment by pbatko-da** [comment]
> Thanks. Additionally, we would need to:
> 
> * [x]  adapt the docs where we mention this restriction at least here (https://docs.daml.com/ops/pruning.html#determine-a-suitable-pruning-offset)
> * [ ]  Remove the redundant insertions in the conformance tests where we do pruning (that were used to ensure pruning before ledger end).

Done the docs update.
I chose not to update the conformance tests as there's a risk that some of the insertions are needed to ensure these tests pass against Canton. Created a ticket to handle it separately from this PR.

**Comment by pbatko-da** [comment]
> Thanks. Additionally, we would need to:
> 
> * [x]  adapt the docs where we mention this restriction at least here (https://docs.daml.com/ops/pruning.html#determine-a-suitable-pruning-offset)
> * [x]  Remove the redundant insertions in the conformance tests where we do pruning (that were used to ensure pruning before ledger end).

Done

**Comment by pbatko-da** [comment]
Won't do for now

**Pull request closed without merging**